app/src/main/java/com/kamron/pogoiv/ShareHandlerActivity.java

```diff
@@ -0,0 +1,37 @@
+package com.kamron.pogoiv;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.v4.content.LocalBroadcastManager;
+import android.os.Bundle;
+
+import java.io.IOException;
+
+public class ShareHandlerActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_handler);
+        finish();
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null && type.startsWith("image/")) {
+            Uri imageUri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
+            try {
+                Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), imageUri);
+                Intent newintent = new Intent("process-bitmap");
+                newintent.putExtra("bitmap", bitmap);
+                LocalBroadcastManager.getInstance(ShareHandlerActivity.this).sendBroadcast(newintent);
```

Comment:
@nahojjjen, the problem is that a lot of the other data in MainActivity would not be initialized. Like trainerlevel, arc calculation caching, tesseract, etc. So, until that's refactored, I've decided not to support this while the app is not running.

Also, after thinking aboit it, I'm not sure we should support this as it quickly starts giving misinformation. I myself scanned old screenshots using the share and I was getting good numbers. I then remembered that those screenshots were when I had a lower trainer level. So. for each share from a saved file, we'll need to add trainer level input.

At least for now or this commit, I didn't feel up to it.